### PR TITLE
Alena burak/fix create topic question

### PR DESCRIPTION
**Pull request by ElenaBurak**

### CODE REVIEW

File: exalearn-api/ExaLearn.WebApi/appsettings.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "DbContext": "Server=localhost;Port=5432;Username=postgres;Password=Pass12345;Database=exalern_db",
+    "DbContext": "Server=localhost;Password=123456;Port=5432;Username=postgres;Password=123;Database=exalern_db",
     "HostingContext": "linkToFilesHosting"
```
Comment (artemgael96) [inline exalearn-api/ExaLearn.WebApi/appsettings.json:12]: return the default settings

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs [hunk 2]
```diff
@@ -44,5 +44,23 @@ public async Task<List<Question>> GetTopicsAsync()
...
+            foreach (var item in topics)
+            {
+                var question = new Question
+                {
+                    QuestionType = QuestionType.Topic,
+                    LevelType = LevelType.Elementary,
+                    Text = item
+                };
+                questions.Add(question);
+            }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs]: ? Use mapper in service

File: exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs [hunk 3]
```diff
@@ -35,13 +35,7 @@ public QuestionProfile()
                 .ForMember(x => x.Question, map => map.MapFrom(source => source.Text))
                 .ForMember(x => x.Answers, map => map.MapFrom(source => source.Answers))
                 .ReverseMap()
-                .ForAllOtherMembers(x => x.Ignore());
-
-            CreateMap<Question, TopicQuestionDTO>()
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs:65]: ?

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 4]
```diff
@@ -43,10 +43,14 @@ public async Task<GrammarQuestionDTO> CreateGrammarQuestionAsync(GrammarQuestion
             return _mapper.Map<GrammarQuestionDTO>(question);
         }
 
-        public async Task<TopicQuestionDTO[]> CreateTopicQuestionAsync(TopicQuestionDTO[] topicQuestionDTO)
+        public async Task<bool> CreateTopicQuestionAsync(TopicQuestionDTO topicQuestionDTO)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: why bool

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 5]
```diff
@@ -43,10 +43,14 @@ public async Task<GrammarQuestionDTO> CreateGrammarQuestionAsync(GrammarQuestion
...
         {
-            var question = await _questionRepository.AddRange(_mapper.Map<Question[]>(topicQuestionDTO));
-            return _mapper.Map<TopicQuestionDTO[]>(question);
+            var result = false;
+            if (topicQuestionDTO.QuestionType == QuestionType.Topic)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: for what

File: exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs (hunk 3 above)
Comment (ExadelSandbox) [reply exalearn-api/ExaLearn.Bl/Mapping/QuestionProfile.cs:65]: maybe missprint

File: exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs [hunk 6]
```diff
@@ -44,5 +44,23 @@ public async Task<List<Question>> GetTopicsAsync()
             Expression<Func<Question, bool>> takeEssayTopic = q => q.QuestionType == QuestionType.Topic;
             return await GetByExpressionAsync(takeEssayTopic, 2);
         }
+
+        //public async Task<bool> AddTopicsAsync(string[] topics)
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.Dal/Repositories/QuestionRepository.cs:48]: if it isn't necessary, delete the commented part of code

File: exalearn-api/ExaLearn.Dal/Interfaces/IQuestionRepository.cs [hunk 7]
```diff
@@ -16,5 +16,7 @@ public interface IQuestionRepository : IGenericRepository<Question>
         public Task<List<Question>> GetAuditionQuestionsAsync(LevelType levelType);
 
         public Task<List<Question>> GetTopicsAsync();
+
+        //public Task<bool> AddTopicsAsync(string[] topics);
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.Dal/Interfaces/IQuestionRepository.cs:20]: what's the reason to leave here commented line

File: exalearn-api/ExaLearn.WebApi/appsettings.json [hunk 8]
```diff
@@ -8,7 +8,7 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "DbContext": "Server=localhost;Port=5432;Username=postgres;Password=Pass12345;Database=exalern_db",
+    "DbContext": "Server=localhost;Password=123456;Port=5432;Username=postgres;Password=123;Database=exalern_db",
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.WebApi/appsettings.json:11]: please remember - NEVER more commit the sensitive information as an user name and a password and NEVER more push it to remote repository. It is strictly prohibited for a real project and can cause your dismiss and penalty. You can leave the sensitive data in your local repository

**Review by ExadelSandbox — CHANGES_REQUESTED** [review]

**Pull request closed without merging**